openshift/console
# ODC-6164: Fix a11y issues in project dropdown

**Fixes**: 
https://issues.redhat.com/browse/ODC-6164

**Analysis / Root cause**: 
1. Missing li around div
2. Missing or invalid `role`s for li elements
3. Some strange "must have sufficient color contrast"

**Solution Description**:
1. Replace `div` with `li` (requires some additional css changes)
2. Change roles from `menuitem` to `option`, similar is used in [PF4 Dropdown](https://www.patternfly.org/v4/components/dropdown/) and add role="none" to dividers
3. Add a white background color to the li elements so that the ace tools doesn't blame the sufficient color contrast anymore.

**Screen shots / Gifs for design review**:

UI is not changed.

A11y error before when project dropdown is closed:

![project-dropdown-closed](https://user-images.githubusercontent.com/139310/126519695-8fd38c03-2061-4ffa-9b46-76a2a7d22e2a.png)

A11y error before when project dropdown is open:

![project-dropdown-open](https://user-images.githubusercontent.com/139310/126519703-4f8218e6-b689-43d6-97a3-5b42d7163b08.png)

A11y error now when project dropdown is open:

![project-dropdown-fixed](https://user-images.githubusercontent.com/139310/126519723-b50e0636-7843-42f3-9789-c9ce2b24f3fb.png)

**Unit test coverage report**: 
Unchanged

**Test setup:**
1. Switch to developer perspective and open the add page
2. Open project dropdown
3. Check a11y errors with Firefox [axe - Web Accessibility Testing](https://addons.mozilla.org/de/firefox/addon/axe-devtools/) extension or by running cypress tests as mentioned in https://issues.redhat.com/browse/ODC-6164

**Browser conformance**:
